LexicalAnalyzer/ExtensionMethods/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using LexicalAnalyzer.Configs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace LexicalAnalyzer.ExtensionMethods
+{
+    public static class IServiceCollectionExtensions
+    {
+        public static IServiceCollection AddLexer<TTokenTypeEnum>(this IServiceCollection services,
+            params TokenDefinition<TTokenTypeEnum>[] tokenDefinitions) where TTokenTypeEnum : Enum
+        { 
+            services.AddTransient<ILexer<TTokenTypeEnum>, Lexer<TTokenTypeEnum>>(provider =>
+            {
+                var lexer = new Lexer<TTokenTypeEnum>();
+
+                foreach (var tokenDefinition in tokenDefinitions)
+                {
+                    lexer.AddTokenDefinition(tokenDefinition.TokenType, tokenDefinition.Pattern);
```

Comment:
Why do you pass the "Pattern" and "TokenType" properties and then create a new "TokenDefinition" instances instead of passing the object itself? And if there is a good reason for that, the function name (AddTokenDefinition) implies it's supposed to get a TokenDefinition instance.

Reply:
It's because I don't want someone from outside change the token definition mid-run.
If it will be a nuget someday I think it should behave like that

Reply:
If this is what they want, why would you care?

Reply:
Because I want it to be immutable, if they want another patterns or somthing they shoudl create another lexer. lexer should never be changed for, it's for their own protection

Reply:
Alright then